### ISSUE TITLE
Post-treatment mean effect

### ISSUE THREAD

**Issue by grantmcdermott** [grantmcdermott/ggiplot#3]
Something to consider once Vincent publishes the next version of **marginaleffects** to CRAN, which exposes the underlying `deltamethod()` function. The basic idea would be to mimic  Figure 7 [here]( https://scholar.harvard.edu/files/shapiro/files/eventstudy.pdf).

Manual example on top of vanilla `iplot()`.

``` r
library(fixest)
library(marginaleffects) # 0.6.0 dev version

es_postaggr = function(object, 
                       rhs = 0, 
                       aggr_type = c("mean", "cumulative"), 
                       abbr_term = TRUE, ...) {
  aggr_type = match.arg(aggr_type)
  fixest_obj = inherits(object, c('fixest', 'fixest_multi'))
  if (!fixest_obj) stop("Please provide a valid fixest object.")
  mm = object$model_matrix_info[[1]]
  if (is.null(mm)) stop ("This function only works with fixest objects that were ",
                         "created with the i() operator. See `?fixest::i()`.")
  coefs = mm$coef_names_full
  ref_id = mm$ref_id[1]
  coefs_post = coefs[(ref_id+1):length(coefs)]
  hypothesis = paste(paste0("`", coefs_post, "`"), collapse = " + ")
  if (aggr_type=="mean") hypothesis = paste0("(", hypothesis, ")/", length(coefs_post))
  hypothesis = paste(hypothesis, "=", rhs)
  ret = marginaleffects::deltamethod(object, hypothesis = hypothesis, ...)
  if (abbr_term) ret$term = paste0("Post treat (", aggr_type, ")")
  attr(ret, "hypothesis") = hypothesis
  return(ret)
}

est = feols(y ~ x1 + i(period, treat, 5) | id + period, base_did)

(post_mean = es_postaggr(est))
#>                term estimate std.error statistic     p.value conf.low conf.high
#> 1 Post treat (mean) 3.906554 0.8598576  4.543257 5.53916e-06 2.221264  5.591844

iplot(est)
polygon(c(5:10, rev(5:10)), 
        c(0, rep(post_mean$conf.low, 5), rev(rep(post_mean$conf.high, 5)), 0),
        col = scales::alpha("orange", .3), lty = 0)
lines(c(rep(0, 5), rep(post_mean$estimate, 5)), lwd = 2, col = "orange")
iplot(est, add = TRUE) # replot to bring errorbars to front
```

![](https://i.imgur.com/OediJtu.png)

<sup>Created on 2022-06-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by grantmcdermott** [comment]
marginaleffects 0.6.0 is now on CRAN 
https://cran.r-project.org/web/packages/marginaleffects/

**Comment by grantmcdermott** [comment]
Closed with #7.
